### DISCUSSION TITLE
useSwiper() is undefined

### DISCUSSION THREAD

**Discussion by braincomb** [nolimits4web/swiper#5504] [Q&A]
I'm trying to use the new `useSwiper` context hooks in Vue, but it's returning undefined:

```js
<script setup lang="ts">
import { Swiper, SwiperSlide, useSwiper } from 'swiper/vue/swiper-vue.js';

const swiper = useSwiper();

swiper.value.update();
</script>
```
`Uncaught TypeError: swiper is undefined`

Using v8.0.6, what am I doing wrong?

Wanted to ask here before opening issue,  @nolimits4web @vltansky any ideas?

**Comment by nolimits4web** [comment]
these hooks must be used in components which are inside of the Swiper

**Reply by braincomb** [reply]
@nolimits4web Thanks for clarifying. Is there no way to get original instance? I'd like to call the update() method since I'm dynamically updating images.

**Reply by nolimits4web** [reply]
Look at docs, you can get it from onSwiper event
